data_models/data_refinery_models/models/test_organisms.py

```diff
@@ -0,0 +1,238 @@
+<!DOCTYPE eEfetchResult PUBLIC "-//NLM//DTD efetch 20131226//EN"
+"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20131226/efetch.dtd">
+<eFetchResult>
+<ERROR>ID list is empty! Possibly it has no correct IDs.</ERROR>
+</eFetchResult>"""
+
+
+def mocked_requests_get(url, parameters):
+    mock = Mock(ok=True)
+    if url is not ESEARCH_URL:
+        mock.text = "This is wrong."
+    else:
+        try:
+            if parameters["field"] is "scin":
+                mock.text = ESEARCH_NOT_FOUND_XML
+            else:
+                mock.text = "This is also wrong."
+        except KeyError:
+            mock.text = ESEARCH_RESPONSE_XML
+
+    return mock
+
+
+class OrganismModelTestCase(TestCase):
+    def tearDown(self):
+        Organism.objects.all().delete()
+
+    @patch('data_refinery_models.models.organism.requests.get')
+    def test_cached_names_are_found(self, mock_get):
+        Organism.objects.create(name="HOMO SAPIENS",
+                                taxonomy_id=9606,
+                                is_scientific_name=True)
+
+        name = Organism.get_name_for_id(9606)
+
+        self.assertEqual(name, "HOMO SAPIENS")
+        mock_get.assert_not_called()
+
+    @patch('data_refinery_models.models.organism.requests.get')
+    def test_cached_ids_are_found(self, mock_get):
+        Organism.objects.create(name="HOMO SAPIENS",
+                                taxonomy_id=9606,
+                                is_scientific_name=True)
+
+        id = Organism.get_id_for_name("Homo Sapiens")
+
+        self.assertEqual(id, 9606)
+        mock_get.assert_not_called()
+
+    @patch('data_refinery_models.models.organism.requests.get')
+    def test_uncached_scientific_names_are_found(self, mock_get):
+        mock_get.return_value = Mock(ok=True)
+        mock_get.return_value.text = ESEARCH_RESPONSE_XML
+
+        taxonomy_id = Organism.get_id_for_name("Homo Sapiens")
+
+        self.assertEqual(taxonomy_id, 9606)
+        mock_get.assert_called_once_with(
+            ESEARCH_URL,
+            {"db": "taxonomy", "field": "scin", "term": "HOMO%20SAPIENS"}
+        )
+
+        # The first call should have stored it in the database so
+        # this call should not make a request.
+        mock_get.reset_mock()
+        new_id = Organism.get_id_for_name("Homo Sapiens")
+
+        self.assertEqual(new_id, 9606)
+        mock_get.assert_not_called()
+
+    @patch('data_refinery_models.models.organism.requests.get')
+    def test_uncached_other_names_are_found(self, mock_get):
+        mock_get.side_effect = mocked_requests_get
+
+        taxonomy_id = Organism.get_id_for_name("Human")
+
+        self.assertEqual(taxonomy_id, 9606)
+        mock_get.assert_has_calls([
+            call(ESEARCH_URL,
+                 {"db": "taxonomy", "field": "scin", "term": "HUMAN"}),
+            call(ESEARCH_URL,
+                 {"db": "taxonomy", "term": "HUMAN"})])
+
+        # The first call should have stored it in the database so
```

Comment:
I suppose "it" here refers the requested organism? Do you mind elaborating this line a little bit to make the comment more readable? Thanks.